lib/streamaggr/streamaggr.go

```diff
@@ -674,6 +686,12 @@ func newAggregator(cfg *Config, pushFunc PushFunc, ms *metrics.Set, opts Options
 		skipIncompleteFlush = !*v
 	}
 
+	minTime := time.Now()
+	if skipIncompleteFlush && alignFlushToInterval {
+		minTime = minTime.Truncate(interval).Add(interval)
+	}
+	a.minTimestamp.Store(minTime.UnixMilli())
```

Comment:
Is this related to the aggregation windows themselves?
Seems like this should address something else, but it is not clear to me what exactly. It would be great to clarify this.

Reply:
it's needed to prevent samples from already flushed window to appear in a same window, which is already waiting for samples from a future